docs/index.rst

```diff
@@ -3,151 +3,717 @@
+        data = Column(Integer, nullable=False, default=0)
+        user_id = Column(Integer, ForeignKey('workflow.id'))
+
+        def __init__(self, grade):
+            self.data = grade
+            session.add(self)
+
+    Base.metadata.create_all(engine)
+
+Next, we have to tell `DbWorkflowEngine` how and when to use our storage. To do
+that we need to know a bit about the engine's `processing_factory` property,
+which is expected to provide this structure of methods and properties:
+
+    * `before_processing`
+    * `after_processing`
+    * `before_object`
+    * `after_object`
+
+    * `action_mapper` (property)
+
+        * `before_callbacks`
+        * `after_callbacks`
+        * `before_each_callback`
+        * `after_each_callback`
+
+    * `transition_exception_mapper` (property)
+
+        * `StopProcessing`
+        * `HaltProcessing`
+        * `ContinueNextToken`
+        * `JumpToken`
+        * `Exception`
+        * ... (Can be extended by adding any method that has the name of an
+          expected exception)
+
+The `transition_exception_mapper` can look confusing at first. It contains not
+Exceptions, but methods that are called when exceptions with the same name are
+raised.
+
+    * Some exceptions are internal to the engine only and never bubble up. (eg
+      `JumpToken`, `ContinueNextToken`)
+    * Others are partly handled internally and then bubbled up to the user to
+      take action. (eg `Exception`)
+
+ Let's use the above to ask our engine to:
+
+    1. Save the first objects that it is given.
+    2. Save to our database every time it finished processing an object and
+           when there is an expected failure.
+
+For now, all we need to know is that in our example, `HaltProcessing` is an
+exception that we will intentionally raise and we want to save the engine when
+it occurs. Once again, follow the comments carefully to understand the code.
+
+.. code-block:: python
+
+    from workflow.engine_db import DbWorkflowEngine
+    from workflow.errors import HaltProcessing
+    from workflow.engine import TransitionActions, ProcessingFactory
+
+    class MyDbWorkflowEngine(DbWorkflowEngine):
+
+        def __init__(self, db_obj):
+            """Load an old `elem_ptr` from the db into the engine."""
+
+            # The reason we save elem_ptr _first_, is because calling `super`
+            # will reset.
+            elem_ptr = db_obj.state_elem_ptr
+
+            self.db_obj = db_obj
+            super(DbWorkflowEngine, self).__init__()
+
+            # And now we inject it back into the engine's `state`.
+            if elem_ptr is not None:
+                self.state.elem_ptr = elem_ptr
+            self.save()
+
+        # For this example we are interested in saving `elem_ptr` as explained
+        # previously, so we override `save` to do that.
+        def save(self, elem_ptr=None):
+            """Save the state of the workflow."""
+            if elem_ptr is not None:
```

Comment:
Its true it would mean a lot of changes for `obj`, indeed. Probably not worth it. 

But since `elem_ptr` is a new kid on the block I feel something close to `token` is better than `element` based which is not mentioned anywhere yet either.

It is meant to represent a position I guess, so what about `token_pos` and `callback_pos` instead of `elem_ptr` and `task_pos`?